### PR TITLE
[JS/WebGPU] Squeeze operator implementation

### PR DESCRIPTION
**Pull request by visheratin**
### Description

This PR adds an implementation of the `Squeeze` operator to WebGPU JSEP. The implementation follows the [operator schema](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Squeeze) and allows one or two inputs.

### How was it tested

1. I created two models. Without `axes`:

```Python
import onnx.helper

node = onnx.helper.make_node(
    "Squeeze",
    inputs=["T"],
    outputs=["y"],
)
graph = onnx.helper.make_graph([node], "test", [onnx.helper.make_tensor_value_info("T", 1, [3, 1, 4, 5])], 
    [onnx.helper.make_tensor_value_info("y", 1, [3, 4, 5])])
onnx.save(onnx.helper.make_model(graph), "squeeze.onnx")
```

And with `axes`:

```Python
import onnx.helper

node = onnx.helper.make_node(
    "Squeeze",
    inputs=["T", "axes"],
    outputs=["y"],
)
graph = onnx.helper.make_graph([node], "test", [onnx.helper.make_tensor_value_info("T", 1, [3, 1, 4, 5]), onnx.helper.make_tensor_value_info("axes", 7, [1])], [onnx.helper.make_tensor_value_info("y", 1, [3, 4, 5])])
onnx.save(onnx.helper.make_model(graph), "squeeze-dim.onnx")
```

2. I compiled the runtime using @fs-eire's [instructions](https://gist.github.com/fs-eire/a55b2c7e10a6864b9602c279b8b75dce).
3. I ran the test models in the browser using this minimal setup:
```HTML
<html>
    <script src=".\dist\ort.webgpu.min.js"></script>
    <script>
        async function run() {
            const session = await ort.InferenceSession.create('squeeze-dim.onnx', {executionProviders: ['webgpu']});
            console.log(session);
            const input = new ort.Tensor('float32', new Float32Array(60), [3, 1, 4, 5]);
            const dim = new ort.Tensor('int64', [-3n], [1]);
            const output = await session.run({ "T": input, "axes": dim });
            console.log(output);
        }
        run();
    </script>
</html>
```

### Motivation and Context

Improve operator coverage for WebGPU JSEP.

### CODE REVIEW

File: onnxruntime/core/providers/js/operators/squeeze.cc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    Squeeze);
+
+ONNX_OPERATOR_VERSIONED_KERNEL_EX(
+    Squeeze,
+    kOnnxDomain,
+    13, 13,
+    kJsExecutionProvider,
+    (*KernelDefBuilder::Create())
+        .TypeConstraint("T", DataTypeImpl::AllFixedSizeTensorTypes())
+        .TypeConstraint("axes", DataTypeImpl::GetTensorType<int64_t>())
```
Comment (satyajandhyala) [inline onnxruntime/core/providers/js/operators/squeeze.cc:28]: Are we only implementing int64_t data type?

File: onnxruntime/core/providers/js/operators/squeeze.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+#pragma once
+
+#include "core/providers/js/js_kernel.h"
+#include "core/framework/data_transfer_manager.h"
+#include "core/providers/cpu/tensor/reshape_helper.h"
+
+namespace onnxruntime {
+namespace js {
+
+class Squeeze final : public JsKernel {
```
Comment (satyajandhyala) [inline onnxruntime/core/providers/js/operators/squeeze.h:13]: Is it possible to derive this from SqueezeBase and inherit functionality?

**Review by satyajandhyala — COMMENTED** [review]
Please check ONNX Squeeze documentation.

**Review by guschmue — APPROVED** [review]

**Comment by fs-eire** [comment]
as suggested in the error message, you may need to run "npm run build:doc" in folder /js/web and checkin the changes of updated file under /js/docs/

**Comment by fs-eire** [comment]
if you have file `generate-operator-md.ts` in your source code, you need to merge latest main branch, where we added `generate-webgpu-operator-md.ts` and js/web/docs/webgpu-operators.md

please make sure do `npm ci` in /js/web before running `npm run build:doc`.

**Review by guschmue — APPROVED** [review]

**Review by fs-eire — APPROVED** [review]

**Comment by guschmue** [comment]
@visheratin ,  thanks for the contribution! 
We sure apricate help with adding ops for webgpu.

**Comment by hariharans29** [comment]
Should this be uncommented now - https://github.com/microsoft/onnxruntime/blob/415c26e46e8fdd1ccc4a39b934de1f9e17bcaa30/js/web/test/suite-test-list.jsonc#L1229 ?

**Comment by fs-eire** [comment]
> Should this be uncommented now -
> 
> https://github.com/microsoft/onnxruntime/blob/415c26e46e8fdd1ccc4a39b934de1f9e17bcaa30/js/web/test/suite-test-list.jsonc#L1229
> 
> ?

good catch... since it's already merged , maybe we should fix it together with a similar implementation of Unsqueeze in another PR?

**Pull request merged**